src/ResponseTagger.php

```diff
@@ -85,7 +106,13 @@ public function hasTags()
      */
     public function addTags(array $tags)
     {
-        $this->tags = array_merge($this->tags, $tags);
+        $filtered = array_filter($tags, 'strlen');
+
+        if ($this->strict && array_diff($filtered, $tags)) {
+            throw new InvalidTagException('Empty tags are not supported');
+        } else {
```

Comment:
no need for the else. in the if you throw an exception, so the execution flow is stopped.